mixer/pkg/runtime/handler/logger.go

```diff
@@ -39,54 +32,56 @@ var _ adapter.Logger = &logger{}
 // logger is used through the adapter.Logger interface.
 const stackDepth = 2
 
-func newLogger(name string) logger {
-	l := log.With(zap.String("adapter", name)).WithOptions(zap.AddCallerSkip(stackDepth))
-	s := l.Sugar()
+var adapterScope = log.RegisterScope("adapters", "Messages from adapters", stackDepth)
 
+type logger struct {
+	name zapcore.Field
+}
+
+func newLogger(name string) logger {
 	return logger{
-		l: l,
-		s: s,
+		name: zap.String("adapter", name),
 	}
 }
 
 // Infof from adapter.Logger.
 func (l logger) Infof(format string, args ...interface{}) {
-	l.s.Infof(format, args...)
+	adapterScope.Info(fmt.Sprintf(format, args...), l.name)
```

Comment:
hmm ... It seems t hat scope should implement the logger API, no?

Reply:
The logger API we're implementing here is adapter-specific, and scope is beyond the mixer's realm.